spec/controllers/groups_controller_spec.rb

```diff
@@ -280,13 +280,99 @@
         end
       end
     end
-    describe '#invalidate_groupings'
-    describe '#validate_groupings'
-    describe '#delete_groupings'
-    describe '#add_members'
-    describe '#add_member'
-    describe '#remove_members'
-    describe '#remove_member'
+
+    describe '#validate_groupings' do
+      let(:grouping) { create :grouping_with_inviter }
+
+      it 'should validate groupings' do
+        post :global_actions, params: {
+          assignment_id: grouping.assignment.id,
+          groupings: [grouping.id],
+          global_actions: 'valid'
+        }
+        expect(grouping.reload.admin_approved).to be true
+      end
+    end
+
+    describe '#invalidate_groupings' do
+      let(:grouping) { create :grouping_with_inviter, admin_approved: true }
+
+      it 'should invalidate groupings' do
+        post :global_actions, params: {
+          assignment_id: grouping.assignment.id,
+          groupings: [grouping.id],
+          global_actions: 'invalid'
+        }
+
+        expect(grouping.reload.admin_approved).to be false
+      end
+    end
+
+    describe '#delete_groupings' do
+      let(:grouping) { create :grouping_with_inviter }
```

Comment:
@robert-tan use `let!` instead of let for these two variables. This should cause them to be evaluated before every test, which should fix the failing tests.